### PR TITLE
TGP-1012: implemented checkTraderProfile

### PR DESCRIPTION
**Pull request by besscerule**

### CODE REVIEW

**Comment by Ewan-James-Donovan** [comment]
Holding off on review until our work is in a happy state :)

**Review by Ewan-James-Donovan — DISMISSED** [review]
Me likey, please don't merge until we've got confirmation on the urls

File: app/connectors/TraderProfileConnector.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  def submitTraderProfile(traderProfile: TraderProfile, eori: String)(implicit hc: HeaderCarrier): Future[Done] =
+    httpClient
+      .put(submitTraderProfileUrl(eori))
+      .withBody(Json.toJson(traderProfile))
+      .execute[HttpResponse]
+      .map(_ => Done)
+
+  private def checkTraderProfileUrl(eori: String) =
+    url"$dataStoreBaseUrl/trader-goods-profiles-data-store/tgp/does-profile-exist/$eori"
```
Comment (Ewan-James-Donovan) [inline app/connectors/TraderProfileConnector.scala]: Now should be /trader-goods-profiles-data-store/profiles/{eori}

File: app/connectors/TraderProfileConnector.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        response.status match {
+          case OK => true
+        }
+      }
+      .recoverWith { case e: NotFoundException =>
+        Future.successful(false)
+      }
+
+  private def getTraderProfileUrl(eori: String) =
+    url"$dataStoreBaseUrl/trader-goods-profiles-data-store/tgp/get-profile/$eori"
```
Comment (Ewan-James-Donovan) [inline app/connectors/TraderProfileConnector.scala]: If we need this, then it has also been renamed: /trader-goods-profiles-data-store/customs/traders/goods-profiles/{eori}

**Review by Ewan-James-Donovan — COMMENTED** [review]
Please see url changes :)

File: app/models/TraderProfile.scala [hunk 3]
```diff
@@ -34,6 +35,7 @@ object TraderProfile {
 
   def build(answers: UserAnswers, eori: String): EitherNec[ValidationError, TraderProfile] =
     (
+      Right(eori),
```
Comment (liandra-giacomine) [inline app/models/TraderProfile.scala]: We actually don't need the EORI in the body, just in the path.

https://developer.service.hmrc.gov.uk/api-documentation/docs/api/service/trader-goods-profiles/1.0/oas/page#tag/Traders-profile/operation/updateTraderGoodsProfile

File: app/connectors/TraderProfileConnector.scala [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  private def checkTraderProfileUrl(eori: String) =
+    url"$dataStoreBaseUrl/trader-goods-profiles-data-store/tgp/does-profile-exist/$eori"
+
+  def checkTraderProfile(eori: String)(implicit hc: HeaderCarrier): Future[Boolean] =
+    httpClient
+      .get(checkTraderProfileUrl(eori))
+      .execute[HttpResponse]
+      .map { response =>
+        response.status match {
```
Comment (liandra-giacomine) [inline app/connectors/TraderProfileConnector.scala:51]: instead of catching an exception, match with the status code:
` case NOT_FOUND => false`

File: app/models/TraderProfile.scala (hunk 3 above)
Comment (besscerule) [reply app/models/TraderProfile.scala]: So this is to make parsing the body we get back easier, as the body we get back has the actor id and the eori in it

Comment (besscerule) [reply app/models/TraderProfile.scala]: changed it and added the data store profile into test

File: app/connectors/TraderProfileConnector.scala [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+class TraderProfileConnector @Inject() (config: Configuration, httpClient: HttpClientV2)(implicit
+  ec: ExecutionContext
+) {
+
+  private val routerBaseUrl: Service    = config.get[Service]("microservice.services.trader-goods-profiles-router")
+  private val dataStoreBaseUrl: Service = config.get[Service]("microservice.services.trader-goods-profiles-data-store")
+
+  private def submitTraderProfileUrl(eori: String) =
```
Comment (liandra-giacomine) [inline app/connectors/TraderProfileConnector.scala]: this should be going to the data store not the router

Comment (liandra-giacomine) [reply app/connectors/TraderProfileConnector.scala]: Path will need to match the data store as discussed in the morning

File: it/test/testModels/DataStoreProfile.scala [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+)
+
+object DataStoreProfile {
+
+  val reads: Reads[DataStoreProfile] = {
+
+    import play.api.libs.functional.syntax._
+
+    (
+      (__ \ "eori").read[String] and
```
Comment (liandra-giacomine) [inline it/test/testModels/DataStoreProfile.scala:36]: just send actorId, eori will be present in the path

Comment (besscerule) [reply it/test/testModels/DataStoreProfile.scala:36]: This is just for the testing - so we can bring back the right shaped data

File: conf/application.conf [hunk 7]
```diff
@@ -50,6 +50,12 @@ microservice {
         port     = 10903
       }
 
+      trader-goods-profiles-data-store {
+        protocol = http
+        host     = localhost
+        port     = 10906
+      }
+
```
Comment (yohan-oc) [inline conf/application.conf:58]: Same config needs to be added here in a separate PR https://github.com/hmrc/app-config-base/blob/5f40d16c52fd5297dcb25adae505beb0b1e0e9d7/trader-goods-profiles-frontend.conf#L16

File: app/views/IndexView.scala.html [hunk 8]
```diff
@@ -1,31 +0,0 @@
...
-
-@layout(
-    pageTitle    = titleNoForm(messages("index.title")),
-    showBackLink = false
-) {
-
-    <h1 class="govuk-heading-l">@messages("index.heading")</h1>
-
-    <p class="govuk-body">@messages("index.guidance")</p>
-}
```
Comment (yohan-oc) [inline app/views/IndexView.scala.html:31]: We can also delete following from messages.en 
`index.title = trader-goods-profiles-frontend
index.heading = trader-goods-profiles-frontend
index.guidance = Welcome to your new frontend. Please see the README file for a guide to getting started.`

Comment (Ewan-James-Donovan) [reply app/views/IndexView.scala.html:31]: agreed

**Review by liandra-giacomine — DISMISSED** [review]

**Review by Ewan-James-Donovan — DISMISSED** [review]

**Comment by Ewan-James-Donovan** [comment]
After discussion, please rename variable routerConnector to something like traderProfileConnector or whatever :)

File: app/connectors/TraderProfileConnector.scala (hunk 5 above)
Comment (besscerule) [reply app/connectors/TraderProfileConnector.scala]: fixed

File: app/views/IndexView.scala.html (hunk 8 above)
Comment (besscerule) [reply app/views/IndexView.scala.html:31]: fixed

**Review by Ewan-James-Donovan — DISMISSED** [review]

**Review by liandra-giacomine — APPROVED** [review]
